arangod/RocksDBEngine/RocksDBSettingsManager.cpp

```diff
@@ -115,183 +110,198 @@ void RocksDBSettingsManager::retrieveInitialValues() {
 }
 
 bool RocksDBSettingsManager::lockForSync(bool force) {
```

Comment:
I think good implementation should make something like strand.
But this is also can be better
```C++
bool RocksDBSettingsManager::lockForSync(bool force) noexcept {
  if (_syncing_mutex.try_lock()) {
    return true;
  }
  if (!force) {
    return false;
  }
  _syncing_mutex.lock();
```
make unlock in scopeGuard

(in really we don't need it as method, and we can return std::unique_lock)

So I really don't understand why we write some spinlocks here